ucsm_apis/admin/callhome.py

```diff
@@ -0,0 +1,397 @@
+    handle.commit()
+    return mo
+
+
+def call_home_contact_update(handle, contact=None, phone=None, email=None,
+                        addr=None, customer=None, contract=None, site=None,
+                        r_from=None, reply_to=None, urgency=None, **kwargs):
+    """
+    Configures call home
+
+    Args:
+        handle (UcsHandle)
+        contact (string): Contact Name
+        phone (string): phone number e.g. +91-1234567890
+        email (string): contact email address
+        addr (string): contact address
+        customer (number): customer id
+        contract (number): contract id
+        site (number): site id
+        r_from (string): from email address
+        reply_to (string): to email address
+        urgency (string): alert priority
+         valid values are "alert", "critical", "debug", "emergency",
+         "error", "info", "notice", "warning"
+        **kwargs: Any additional key-value pair of managed object(MO)'s
+                  property and value, which are not part of regular args.
+                  This should be used for future version compatibility.
+    Returns:
+        SmartcallhomeSource: ManagedObject
+
+    Raises:
+        UcsOperationError: If SmartcallhomeSource is not present
+
+    Example:
+        from ucsmsdk.mometa.callhome.SmartcallhomeSource import \
+            SmartcallhomeSourceConsts
+
+        call_home_config(handle,
+                         contact="user name",
+                         phone="+91-1234567890",
+                         email="user@cisco.com",
+                         addr="user address",
+                         customer="1111",
+                         contract="2222",
+                         site="3333",
+                         r_from="from@cisco.com",
+                         reply_to="to@cisco.com",
+                         urgency=CallhomeSourceConsts.URGENCY_ALERT,
+                         )
+    """
+
+    # configure call home
+    dn = _base_dn + "/source"
+    mo = handle.query_dn(dn)
+    if not mo:
+        raise UcsOperationError("call_home_config",
+                                 "Call home source '%s' not available." % dn)
+
+    args = {'contact': contact,
+            'phone': phone,
+            'email': email,
+            'addr': addr,
+            'customer': customer,
+            'contract': contract,
+            'site': site,
+            'r_from': r_from,
+            'reply_to': reply_to,
+            'urgency': urgency}
+
+    mo.set_prop_multiple(**args)
+    mo.set_prop_multiple(**kwargs)
+
+    handle.set_mo(mo)
+    handle.commit()
+    return mo
+
+def call_home_smtp_update(handle, host=None, port="25", **kwargs):
+    dn = _base_dn + "/smtp"
+    mo = handle.query_dn(dn)
+    if not mo:
+        raise UcsOperationError("call_home_smtp_update",
+                            "Call home smtp server '%s' not available." % dn)
+
+    args = {'host': host, 'port': port}
+
+    mo.set_prop_multiple(**args)
```

Comment:
if `host=None` is passed to `set_prop_multiple` and in turn to `__set_prop` it will fail during property validation as `None` is not a valid value